src/util.js

```diff
@@ -106,3 +107,127 @@ export function format_memory_and_limit(usage, limit) {
         return _("");
     }
 }
+
+// TODO
+export function container_stop(container, timeout) {
+}
+
+export function updateContainers() {
+    let newContainers = [];
+    let newContainersStats = [];
+    return new Promise((resolve, reject) => {
+        varlinkCall(PODMAN, "io.podman.ListContainers")
+                .then(reply => {
+                    let newContainersMeta = reply.containers;
+                    if (!newContainersMeta) {
+                        resolve({newContainers: newContainers, newContainersStats: newContainersStats});
+                        return;
+                    }
+                    const containerCount = newContainersMeta.length;
+                    let inspectRet = [];
+                    newContainersMeta.map((container) => {
+                        let proEle = new Promise((resolve, reject) => {
+                            varlinkCall(PODMAN, "io.podman.InspectContainer", {name: container.id})
+                                    .then(reply => {
+                                        resolve(JSON.parse(reply.container));
+                                    })
+                                    .catch(ex => {
+                                        reject(new Error("Failed to do InspectContainer call:", ex, JSON.stringify(ex)));
+                                    });
+                        });
+                        inspectRet.push(proEle);
+                    });
+                    Promise.all(inspectRet)
+                            .then((inspectRet) => {
+                                let runEle = newContainersMeta.filter((ele) => {
+                                    return ele.status === "running";
+                                });
+                                inspectRet.map((inspectRet) => {
+                                    newContainers[inspectRet.ID] = inspectRet;
+                                    if (runEle.length === 0 && Object.keys(newContainers).length === containerCount) {
+                                        resolve({newContainers: newContainers, newContainersStats: newContainersStats});
+                                    }
```

Comment:
This is a strange special case. I see what you want to do with the `length === containerCount` (resolve the promise after the last iteration), but this is absolutely not necessary to do it in this convoluted way. Just move the `resolve()` after the `map()` call.

Reply:
Add more promise chain in [#18](https://github.com/cockpit-project/cockpit-podman/pull/18) and move it out of map()